src/sentry/static/sentry/app/stores/alertStore.jsx

```diff
@@ -9,34 +10,43 @@ const AlertStore = Reflux.createStore({
     this.count = 0;
   },
 
-  onAddAlert(message, type, expireAfter, url) {
-    // intentionally recreate array via concat because of Reflux
-    // "bug" where React components are given same reference to tracked
-    // data objects, and don't *see* that values have changed
-    let alertId = this.count++;
-
-    this.alerts = this.alerts.concat([{
-      id: alertId,
-      message: message,
-      type: type,
-      url: url
-    }]);
+  getMutedKey(alert) {
+    return `alerts:${alert.id}:muted`;
+  },
 
-    if (typeof expireAfter === 'undefined') {
-      expireAfter = 5000;
+  onAddAlert(alert) {
+    if (alert.id !== void 0) {
+      if (getItem(this.getMutedKey(alert)) !== null) {
+        return;
+      }
+    } else {
+      if (alert.expireAfter === void 0) {
+        alert.expireAfter = 5000;
+      }
     }
-    if (expireAfter) {
+
+    if (alert.expireAfter) {
       window.setTimeout(() => {
-        this.onCloseAlert(alertId);
-      }, expireAfter);
+        this.onCloseAlert(alert);
+      }, alert.expireAfter);
     }
 
+    alert.key = this.count++;
+
+    // intentionally recreate array via concat because of Reflux
+    // "bug" where React components are given same reference to tracked
+    // data objects, and don't *see* that values have changed
+    this.alerts = this.alerts.concat([alert]);
     this.trigger(this.alerts);
   },
 
-  onCloseAlert(id) {
+  onCloseAlert(alert) {
+    if (alert.id !== void 0) {
+      setItem(this.getMutedKey(alert), +new Date());
```

Comment:
The problem here is this key will never get cleaned out of LocalStorage. It doesn't have expiration, and there is a limited amount of space that can be used. So it's not great if this never got removed.

And, since the keys are leveraging the md5 hash, I don't think it's even possible to iterate them all and check. I think typically this would just use one big key, and store JSON to represent all the muted things, then you could apply other cleanup logic, etc.